res/css/views/right_panel/_RoomSummaryCard.pcss

```diff
@@ -42,17 +42,19 @@ limitations under the License.
         .mx_RoomSummaryCard_avatar {
             display: inline-flex;
 
-            .mx_RoomSummaryCard_e2ee {
+            .mx_RoomSummaryCard_avatar_e2ee {
                 display: inline-block;
                 position: relative;
                 width: 54px;
                 height: 54px;
                 border-radius: 50%;
-                background-color: #737d8c;
+                background-color: var(--backgroundColor);
                 margin-top: -3px; /* alignment */
                 margin-left: -10px; /* overlap */
                 border: 3px solid $dark-panel-bg-color;
 
+                --backgroundColor: #737d8c;
```

Comment:
what is the value of extracting a variable here?

(likewise below)

Reply:
The Sass variable will eventually be replaced by CSS custom property, and I thought this would make it work in a better way.

Do you want me to split that task into another PR, as the title of this PR indeed does not indicate the replacement?

Reply:
sorry, when I said "variable", I meant "custom property".

Why is it better to define a `--backgroundColour` custom property than to use the standard `background-color` property?

Reply:
Because it should make it possible to optimize the rules, using the [cascade](https://css-tricks.com/a-complete-guide-to-custom-properties/#aa-using-the-cascade).

Here is a simple example with Sass variables;
````
.card {
    background-color: $card-red;

    &.card--blue {
        background-color: $card-blue;
    }
}
````

With custom properties, if you would want to keep those two declarations, you would have to write like this:
````
.card {
    --color: $card-red;
    background-color: var(--color);

    &.card--blue {
        --color: $card-blue;
        background-color: var(--color);
    }
}
````
The second `background-color: var(--color)` is not necessary and can be removed, since `.card.card--blue` has higher specificity than `.card`, and the value of the custom property inside `.card--blue`, which is `$card-blue`, overrides `$card-red` thanks to the cascade.

Therefore, this can be rewritten like this:

````
.card {
    --color: $card-red; /* applied by default */
    background-color: var(--color);

    &.card--blue {
        --color: $card-blue; /* `$card-blue` is applied as background-color of `.card.card-blue` because this has higher specificity than `.card` */
    }
}
````

Reply:
Here is a good example: https://github.com/matrix-org/matrix-react-sdk/commit/6b3098d8fe0cfd1633f5f9474030650337fa2a60#diff-2619500f973975a4b82b359c305c5d07f9f76b95e06abbce0d3486f3a7436cb1R20-R64

Reply:
Thank you for explaining. I understood your point, and I think that makes sense. I am going to revert the change.

I think your argument should be worth being applied to our style guide to define the usage of custom properties, and I am wondering if you could take some time to do so…

Reply:
> I think your argument should be worth being applied to our style guide to define the usage of custom properties, and I am wondering if you could take some time to do so…

Honestly, if I were to try to write this down for the style guide, it would be "do not create custom properties with no value". I think if we try to add rules like that to the style guide, the style guide will become 1000 pages long.

Reply:
(In case you might be misunderstanding, I was not being cynical here at all… having a chapter on custom properties usage should generally help us all IMO,)